### PR TITLE
Update TTS module’s docstrings and refactor some modules

### PR DESCRIPTION
**Pull request by kan-bayashi**
This PR mainly updates docstrings of TTS modules to follow the gooogle style + flake8-docstrings.

Related #876 #894
 
- fix docstrings 
  - [x] `espnet/nets/pytorch_backend/e2e_tts_tacotron2`
  - [x] `espnet/nets/pytorch_backend/e2e_tts_transformer`
  - [x] `espnet/nets/pytorch_backend/e2e_tts_fastspeech`
  - [x] `espnet/nets/pytorch_backend/tacotron2/encoder.py`
  - [x] `espnet/nets/pytorch_backend/tacotron2/decoder.py`
  - [x] `espnet/nets/pytorch_backend/tacotron2/cbhg.py`
  - [x] `espnet/nets/pytorch_backend/transformer/multi_layer_conv.py`
  - [x] `espnet/nets/pytorch_backend/fastspeech/duration_calculator.py`
  - [x] `espnet/nets/pytorch_backend/fastspeech/duration_predictor.py`
  - [x] `espnet/nets/pytorch_backend/fastspeech/length_regulator.py`
  - [x] `espnet/nets/pytorch_backend/nets_utils.py`
- move some functions in `nets_uitls.py` as private method of `rnn.Decoder` because these functions are used only once for specific method of `rnn.Decoder`.
- delete never used function `nets_utils.expand_yseq`
- add `grrifin_lim_iters` option in `egs/ljspeech/tts2/run.sh`. I wrongly delete it in #780 due to my misunderstanding.

### CODE REVIEW

**Comment by kan-bayashi** [comment]
Finished.
I will continue to update docstrings in the next PR.

**Comment by ShigekiKarita** [comment]
Is it possible to remove these warnings?
https://travis-ci.org/espnet/espnet/jobs/548724605#L1954-L1961

**Comment by kan-bayashi** [comment]
Sphinx == 1.7.4 has a problem with napoleon.
https://github.com/sphinx-doc/sphinx/issues/5076
I update the Sphinx version to solve this problem.

**Comment by kan-bayashi** [comment]
I confirmed warnings are removed.

**Comment by ShigekiKarita** [comment]
GREAT.

Now you should remove this `--nowarn`. CI can notify us on sphinx warnings (typically syntax error) as a failure.
https://github.com/kan-bayashi/espnet/blob/google_style_doc/ci/doc.sh#L25

**Comment by kan-bayashi** [comment]
Sorry, not modified part in this PR still has warning.
We should fix this in the next PR before remove `--nowarn`.

```
/Users/mo-chi/work/espnet/espnet/lm/lm_utils.py:docstring of espnet.lm.lm_utils.count_tokens:5: WARNING: Field list ends without a blank line; unexpected unindent.
/Users/mo-chi/work/espnet/espnet/lm/pytorch_backend/lm.py:docstring of espnet.lm.pytorch_backend.lm.ClassifierWithState.final:4: WARNING: Field list ends without a blank line; unexpected unindent.
/Users/mo-chi/work/espnet/espnet/lm/pytorch_backend/lm.py:docstring of espnet.lm.pytorch_backend.lm.concat_examples:7: WARNING: Field list ends without a blank line; unexpected unindent.
/Users/mo-chi/work/espnet/espnet/nets/chainer_backend/attentions.py:docstring of espnet.nets.chainer_backend.attentions.att_for:5: WARNING: Field list ends without a blank line; unexpected unindent.
/Users/mo-chi/work/espnet/espnet/nets/chainer_backend/nets_utils.py:docstring of espnet.nets.chainer_backend.nets_utils.linear_tensor:6: WARNING: Field list ends without a blank line; unexpected unindent.
/Users/mo-chi/work/espnet/espnet/nets/pytorch_backend/ctc.py:docstring of espnet.nets.pytorch_backend.ctc.ctc_for:4: WARNING: Field list ends without a blank line; unexpected unindent.
/Users/mo-chi/work/espnet/espnet/nets/pytorch_backend/e2e_asr_mix.py:docstring of espnet.nets.pytorch_backend.e2e_asr_mix.PIT.min_pit_sample:5: WARNING: Field list ends without a blank line; unexpected unindent.
/Users/mo-chi/work/espnet/espnet/nets/pytorch_backend/e2e_asr_mix.py:docstring of espnet.nets.pytorch_backend.e2e_asr_mix.PIT.pit_process:4: WARNING: Field list ends without a blank line; unexpected unindent.
/Users/mo-chi/work/espnet/espnet/nets/pytorch_backend/e2e_asr_transformer.py:docstring of espnet.nets.pytorch_backend.e2e_asr_transformer.subsequent_mask:7: WARNING: Field list ends without a blank line; unexpected unindent.
/Users/mo-chi/work/espnet/espnet/tts/pytorch_backend/tts.py:docstring of espnet.tts.pytorch_backend.tts.CustomUpdater:4: WARNING: Field list ends without a blank line; unexpected unindent.
/Users/mo-chi/work/espnet/espnet/utils/spec_augment.py:docstring of espnet.utils.spec_augment.apply_interpolation:8: WARNING: Unexpected indentation.
/Users/mo-chi/work/espnet/espnet/utils/spec_augment.py:docstring of espnet.utils.spec_augment.apply_interpolation:9: WARNING: Block quote ends without a blank line; unexpected unindent.
/Users/mo-chi/work/espnet/espnet/utils/spec_augment.py:docstring of espnet.utils.spec_augment.interpolate_bilinear:9: WARNING: Unexpected indentation.
/Users/mo-chi/work/espnet/espnet/utils/spec_augment.py:docstring of espnet.utils.spec_augment.interpolate_bilinear:10: WARNING: Block quote ends without a blank line; unexpected unindent.
/Users/mo-chi/work/espnet/espnet/utils/spec_augment.py:docstring of espnet.utils.spec_augment.interpolate_bilinear:14: WARNING: Unexpected indentation.
```

File: espnet/nets/pytorch_backend/e2e_tts_tacotron2.py [hunk 1]
```diff
@@ -23,14 +23,18 @@
...
-        Efficiently Trainable Text-to-Speech System Based on Deep Convolutional Networks with Guided Attention
-        (https://arxiv.org/abs/1710.08969)
+    This module calculates the guided attention loss described in `Efficiently Trainable Text-to-Speech System Based
+    on Deep Convolutional Networks with Guided Attention`_, which forces the attention to be diagonal.
+
+    Args:
+        sigma (float, optional): Standard deviation to control how close attention to a diagonal.
+        reset_always (bool, optional): Whether to always reset masks.
+
+    .. _`Efficiently Trainable Text-to-Speech System Based on Deep Convolutional Networks with Guided Attention`:
```
Comment (ShigekiKarita) [inline espnet/nets/pytorch_backend/e2e_tts_tacotron2.py]: use `References` section directive? http://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#confval-napoleon_use_admonition_for_references

File: espnet/nets/pytorch_backend/e2e_tts_fastspeech.py [hunk 2]
```diff
@@ -28,50 +28,54 @@
...
+            - transformer_enc_attn_dropout_rate (float): Dropout rate in encoder self-attention module.
+            - transformer_dec_dropout_rate (float): Dropout rate in decoder except for attention & positional encoding.
+            - transformer_dec_positional_dropout_rate (float): Dropout rate after decoder positional encoding.
+            - transformer_dec_attn_dropout_rate (float): Dropout rate in deocoder self-attention module.
+            - transformer_enc_dec_attn_dropout_rate (float): Dropout rate in encoder-deocoder attention module.
+            - use_masking (bool): Whether to use masking in calculation of loss.
+            - transfer_encoder_from_teacher: Whether to transfer encoder using teacher encoder parameters.
+            - transferred_encoder_module: Encoder module to be initialized using teacher parameters.
+
+    .. _`FastSpeech: Fast, Robust and Controllable Text to Speech`:
```
Comment (ShigekiKarita) [inline espnet/nets/pytorch_backend/e2e_tts_fastspeech.py]: ditto

File: espnet/nets/pytorch_backend/fastspeech/duration_predictor.py [hunk 3]
```diff
@@ -10,18 +10,26 @@
...
+
+    Args:
+        idim (int): Input dimension.
+        n_layers (int, optional): Number of convolutional layers.
+        n_chans (int, optional): Number of channels of convolutional layers.
+        kernel_size (int, optional): Kernel size of convolutional layers.
+        dropout_rate (float, optional): Dropout rate.
+        offset (float, optional): Offset value to avoid nan in log domain.
+
+    .. _`FastSpeech: Fast, Robust and Controllable Text to Speech`:
```
Comment (ShigekiKarita) [inline espnet/nets/pytorch_backend/fastspeech/duration_predictor.py]: ditto

File: espnet/nets/pytorch_backend/fastspeech/length_regulator.py [hunk 4]
```diff
@@ -10,30 +10,36 @@
...
+    The length regulator expands char or phoneme-level embedding features to frame-level by repeating each
+    feature based on the corresponding predicted durations.
 
-    Reference:
-        FastSpeech: Fast, Robust and Controllable Text to Speech
-        (https://arxiv.org/pdf/1905.09263.pdf)
+    Args:
+        pad_value (float, optional): Value used for padding.
+
+    .. _`FastSpeech: Fast, Robust and Controllable Text to Speech`:
```
Comment (ShigekiKarita) [inline espnet/nets/pytorch_backend/fastspeech/length_regulator.py]: ditto

File: espnet/nets/pytorch_backend/tacotron2/cbhg.py [hunk 5]
```diff
@@ -11,21 +11,24 @@
...
+        odim (int): Dimension of the outputs.
+        conv_bank_layers (int, optional): The number of convolution bank layers.
+        conv_bank_chans (int, optional): The number of channels in convolution bank.
+        conv_proj_filts (int, optional): Kernel size of convolutional projection layer.
+        conv_proj_chans (int, optional): The number of channels in convolutional projection layer.
+        highway_layers (int, optional): The number of highway network layers.
+        highway_units (int, optional): The number of highway network units.
+        gru_units (int, optional): The number of GRU units (for both directions).
+
+    .. _`Tacotron: Towards End-to-End Speech Synthesis`: https://arxiv.org/abs/1703.10135
```
Comment (ShigekiKarita) [inline espnet/nets/pytorch_backend/tacotron2/cbhg.py]: ditto

File: espnet/nets/pytorch_backend/transformer/multi_layer_conv.py [hunk 6]
```diff
@@ -8,16 +8,20 @@
...
+    This is a module of multi-leyered conv1d designed to replace positionwise feed-forward network
+    in Transforner block, which is introduced in `FastSpeech: Fast, Robust and Controllable Text to Speech`_.
+
+    Args:
+        in_chans (int): Number of input channels.
+        hidden_chans (int): Number of hidden channels.
+        kernel_size (int): Kernel size of conv1d.
+        dropout_rate (float): Dropout rate.
+
+    .. _`FastSpeech: Fast, Robust and Controllable Text to Speech`:
```
Comment (ShigekiKarita) [inline espnet/nets/pytorch_backend/transformer/multi_layer_conv.py]: ditto

File: espnet/nets/pytorch_backend/e2e_tts_tacotron2.py [hunk 7]
```diff
@@ -212,53 +231,59 @@ def forward(self, after_outs, before_outs, logits, ys, labels, olens):
...
+            - cbhg_conv_bank_chans (int): The number of channels of convolutional bank in CBHG.
+            - cbhg_proj_filts (int): The number of filter size of projection layeri in CBHG.
+            - cbhg_proj_chans (int): The number of channels of projection layer in CBHG.
+            - cbhg_highway_layers (int): The number of layers of highway network in CBHG.
+            - cbhg_highway_units (int): The number of units of highway network in CBHG.
+            - cbhg_gru_units (int): The number of units of GRU in CBHG.
+            - use_masking (bool): Whether to mask padded part in loss calculation.
+            - bce_pos_weight (float): Weight of positive sample of stop token (only for use_masking=True).
+
+    .. _`Natural TTS Synthesis by Conditioning WaveNet on Mel Spectrogram Predictions`:
```
Comment (ShigekiKarita) [inline espnet/nets/pytorch_backend/e2e_tts_tacotron2.py]: ditto

File: espnet/nets/pytorch_backend/e2e_tts_transformer.py [hunk 8]
```diff
@@ -130,61 +143,66 @@ def plotfn(self, data, attn_dict, outdir, suffix="png", savefn=None):
...
+            - postnet_dropout_rate (float): Dropout rate in postnet.
+            - use_masking (bool): Whether to use masking in calculation of loss.
+            - bce_pos_weight (float): Positive sample weight in bce calculation (only for use_masking=true).
+            - loss_type (str): How to calculate loss.
+            - use_guided_attn_loss (bool): Whether to use guided attention loss.
+            - num_heads_applied_guided_attn (int): Number of heads in each layer to be applied guided attention loss.
+            - num_layers_applied_guided_attn (int): Number of layers to be applied guided attention loss.
+            - modules_applied_guided_attn (list): List of module names to be applied guided attention loss.
+
+    .. _`Neural Speech Synthesis with Transformer Network`:
```
Comment (ShigekiKarita) [inline espnet/nets/pytorch_backend/e2e_tts_transformer.py]: ditto

**Comment by ShigekiKarita** [comment]
> Sorry, not modified part in this PR still has warning.
> We should fix this in the next PR before remove `--nowarn`.
> 
> ```
> /Users/mo-chi/work/espnet/espnet/lm/lm_utils.py:docstring of espnet.lm.lm_utils.count_tokens:5: WARNING: Field list ends without a blank line; unexpected unindent.
> /Users/mo-chi/work/espnet/espnet/lm/pytorch_backend/lm.py:docstring of espnet.lm.pytorch_backend.lm.ClassifierWithState.final:4: WARNING: Field list ends without a blank line; unexpected unindent.
> /Users/mo-chi/work/espnet/espnet/lm/pytorch_backend/lm.py:docstring of espnet.lm.pytorch_backend.lm.concat_examples:7: WARNING: Field list ends without a blank line; unexpected unindent.
> /Users/mo-chi/work/espnet/espnet/nets/chainer_backend/attentions.py:docstring of espnet.nets.chainer_backend.attentions.att_for:5: WARNING: Field list ends without a blank line; unexpected unindent.
> /Users/mo-chi/work/espnet/espnet/nets/chainer_backend/nets_utils.py:docstring of espnet.nets.chainer_backend.nets_utils.linear_tensor:6: WARNING: Field list ends without a blank line; unexpected unindent.
> /Users/mo-chi/work/espnet/espnet/nets/pytorch_backend/ctc.py:docstring of espnet.nets.pytorch_backend.ctc.ctc_for:4: WARNING: Field list ends without a blank line; unexpected unindent.
> /Users/mo-chi/work/espnet/espnet/nets/pytorch_backend/e2e_asr_mix.py:docstring of espnet.nets.pytorch_backend.e2e_asr_mix.PIT.min_pit_sample:5: WARNING: Field list ends without a blank line; unexpected unindent.
> /Users/mo-chi/work/espnet/espnet/nets/pytorch_backend/e2e_asr_mix.py:docstring of espnet.nets.pytorch_backend.e2e_asr_mix.PIT.pit_process:4: WARNING: Field list ends without a blank line; unexpected unindent.
> /Users/mo-chi/work/espnet/espnet/nets/pytorch_backend/e2e_asr_transformer.py:docstring of espnet.nets.pytorch_backend.e2e_asr_transformer.subsequent_mask:7: WARNING: Field list ends without a blank line; unexpected unindent.
> /Users/mo-chi/work/espnet/espnet/tts/pytorch_backend/tts.py:docstring of espnet.tts.pytorch_backend.tts.CustomUpdater:4: WARNING: Field list ends without a blank line; unexpected unindent.
> /Users/mo-chi/work/espnet/espnet/utils/spec_augment.py:docstring of espnet.utils.spec_augment.apply_interpolation:8: WARNING: Unexpected indentation.
> /Users/mo-chi/work/espnet/espnet/utils/spec_augment.py:docstring of espnet.utils.spec_augment.apply_interpolation:9: WARNING: Block quote ends without a blank line; unexpected unindent.
> /Users/mo-chi/work/espnet/espnet/utils/spec_augment.py:docstring of espnet.utils.spec_augment.interpolate_bilinear:9: WARNING: Unexpected indentation.
> /Users/mo-chi/work/espnet/espnet/utils/spec_augment.py:docstring of espnet.utils.spec_augment.interpolate_bilinear:10: WARNING: Block quote ends without a blank line; unexpected unindent.
> /Users/mo-chi/work/espnet/espnet/utils/spec_augment.py:docstring of espnet.utils.spec_augment.interpolate_bilinear:14: WARNING: Unexpected indentation.
> ```

Oh. That is OK.

File: espnet/nets/pytorch_backend/e2e_tts_tacotron2.py (hunk 1 above)
Comment (kan-bayashi) [reply espnet/nets/pytorch_backend/e2e_tts_tacotron2.py]: What do you mean?
use `napoleon_use_admonition_for_references` option?
e.g.
```python
extensions = [
    "sphinx.ext.autodoc",
    'sphinx.ext.napoleon',
    'sphinx.ext.viewcode',
    "sphinx.ext.mathjax",
    "sphinxarg.ext",
]

# napoleon setting
napoleon_use_admonition_for_references = True
````

Comment (ShigekiKarita) [reply espnet/nets/pytorch_backend/e2e_tts_tacotron2.py]: Sorry, I misunderstood what the options is.

**Comment by ShigekiKarita** [comment]
LGTM

**Pull request merged**